ext/compressors/lz4/lz4_compress.c

```diff
@@ -48,19 +48,38 @@ typedef struct {
 } LZ4_COMPRESSOR;
 
 /*
+ * LZ4 decompression requires the exact compressed byte count returned by the
+ * LZ4_compress and LZ4_compress_destSize functions. WiredTiger doesn't track
+ * that value, store it in the destination buffer.
+ *
+ * Additionally, LZ4_compress_destSize may compress into the middle of a record,
+ * and after decompression we return the length to the last record successfully
+ * decompressed, not the number of bytes decompressed; store that value in the
+ * destination buffer as well.
+ *
+ * Use fixed-size, 4B values (WiredTiger never writes buffers larger than 4GB).
+ */
+typedef struct {
+	uint32_t compressed_len;	/* True compressed length */
+	uint32_t compressed_src;	/* True uncompressed source length */
```

Comment:
I did it that way because the `compress` function is the one that sets the value, but I see your point. I've switched to:
```
typedef struct {
        uint32_t compressed_len;        /* True compressed length */
        uint32_t uncompressed_len;      /* True uncompressed source length */
        uint32_t useful_len;            /* Decompression return value */
        uint32_t unused;
} LZ4_PREFIX;
```
hopefully that's better.